From 812d97a8c71f6014d4c2386632c08cc6095a8860 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Wed, 17 Oct 2012 16:02:04 +0530
Subject: [PORT FROM MAIN]Playback of the mp4_he_aac_v2 file is not supported (Metadata corrupted; erroraneous content).

BZ: 59738

In current MPEG4 Extractor, error is returned if number of channels are zero.
Due to which playback of some files is not supported.
With this fix, removing the check for numChannels from MPEG4Extractor, as this
condition (zero Channels) has been handled in UMCDecoder.h.

Change-Id: Iffc5d466adb6f7da37f0c08ef3bef7e1a9cfe438
Orig-Change-Id: I8c3b882494b3b8242a6f5ae8f6fb3edab7c2d800
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 606b407..25d4610 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -2376,9 +2376,6 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
         }
     }
 
-    if (numChannels == 0) {
-        return ERROR_UNSUPPORTED;
-    }
 
     int32_t prevSampleRate;
     CHECK(mLastTrack->meta->findInt32(kKeySampleRate, &prevSampleRate));
-- 
1.7.4.1

